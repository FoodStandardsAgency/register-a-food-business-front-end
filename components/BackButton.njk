{% from "../node_modules/govuk-frontend/dist/govuk/components/back-link/macro.njk" import govukBackLink %}
{% macro backButton(props) %}
  {% if not(props.hideback or props.editModeFirstPage) %}
    {% if props.backToStart %}
      {{ govukBackLink({
              href: props.href if props.href else "/new",
              text: __("Back to start", props.language)
          }) }}
    {% elif props.currentPage == "/partner-details" %}
      {{ govukBackLink({
              href: props.partnerDetailsBackUrl,
              text: __("Back", props.language)
          }) }}
    {% elif props.currentPage == "/la-selector" %}
      {% if props.fullCurrentPage === "/la-selector?back=wrong-la" %}
        {{ govukBackLink({
              href: "/new/la-established",
              text: __("Back", props.language)
          }) }}
      {% else %}
        {{ govukBackLink({
              href: "/new/establishment-address",
              text: __("Back", props.language)
          }) }}
      {% endif %}
      
    {% else %}
      {{ govukBackLink({
              href: props.backHref if props.backHref else ("/back" + props.currentPage),
              text: __("Back", props.language)
          }) }}
    {% endif %}
  {% endif %}
{% endmacro %}