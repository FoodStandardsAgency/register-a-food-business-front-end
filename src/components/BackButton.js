//import { BackLink } from "govuk-react";

const BackButton = props => (
  <a href={"/back/" + props.originator} id="back-link">
    Back
  </a>
);
export default BackButton;
