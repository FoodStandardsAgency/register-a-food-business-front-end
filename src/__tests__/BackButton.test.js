import BackButton from "../components/BackButton";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("<BackButton />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<BackButton />);
    expect(wrapper.length).toBe(1);
  });

  it("matches the previous snapshot", () => {
    const tree = renderer.create(<BackButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("gets given the correct originator props", () => {
    const originator = "previous-page";

    const wrapper = mount(<BackButton originator={originator} />);
    const backButton = wrapper.find("BackButton");
    expect(backButton.props().originator).toBe(originator);
  });

  it("gets given the correct editModeFirstPage props", () => {
    const editModeFirstPage = "/page";

    const wrapper = mount(<BackButton editModeFirstPage={editModeFirstPage} />);
    const backButton = wrapper.find("BackButton");
    expect(backButton.props().editModeFirstPage).toBe(editModeFirstPage);
  });

  it("It displays no content when given that editModeFirstPage is the same as currentPage", () => {
    const editModeFirstPage = "/page";
    const currentPage = "/page";

    const wrapper = mount(
      <BackButton
        currentPage={currentPage}
        editModeFirstPage={editModeFirstPage}
      />
    );
    const backElement = wrapper.find(`a#back-link`);
    expect(backElement.length).toBe(0);
  });

  it("It displays when given that editModeFirstPage is NOT the same as currentPage", () => {
    const editModeFirstPage = "/page";
    const currentPage = "/another";

    const wrapper = mount(
      <BackButton
        currentPage={currentPage}
        editModeFirstPage={editModeFirstPage}
      />
    );
    const backElement = wrapper.find(`a#back-link`);
    expect(backElement.length).toBe(1);
  });

  it("passes the given custom href if provided", () => {
    const wrapper = mount(<BackButton href="example" />);
    const backElement = wrapper.find(`a#back-link`);
    expect(backElement.props().href).toBe("example");
  });
});
