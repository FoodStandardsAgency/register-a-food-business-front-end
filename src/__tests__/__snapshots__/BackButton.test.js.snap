// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> matches the previous snapshot 1`] = `
<a
  href="/back/undefined"
  id="back-link"
>
  Back
</a>
`;
